### DISCUSSION TITLE
issue:incorrect validation on ipad/iphone

### DISCUSSION THREAD

**Discussion by ferrybrouwer** [react-hook-form/react-hook-form#7067] [Q&A]
### Version Number

7.19.5

### Codesandbox/Expo snack

https://codesandbox.io/s/nifty-cerf-bjev5?file=/src/App.jsx

### Just in chrome (no simulation):

1. Go to https://bjev5.csb.app/
2. Select an item from the dropdown (react-select)
3. click the submit button
3. clear the item from dropdown by pressing X in the dropdown
4. inspect there's an error message "can not be empty"
5. now add the item again (the error message will dissapear)
6. click the submit button again
7. notice the error message is gone (like it should)

### Chrome ipad simulation (devtools):

Do the exact same steps as described above. 
You'll notice Step 7 gives a different output; the error message is still being displayed!!

### Expected behaviour

I would expect Step 7 will not be different on ipad/iphone.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> Note: This config aligns with web constrained API for required input validation, for object or array type of input value use validate function instead.

refer to the doc on register required attribute

**Reply by ferrybrouwer** [reply]
This is actually not the problem. Somehow the value is not set properly. 

When you do `getValues()` within an interval the values are just not updated properly.

See https://codesandbox.io/s/nifty-cerf-bjev5?file=/src/App.jsx 

1. open https://bjev5.csb.app/ this in chrome 
2. open devtools simulate ipad
3. open console to inspect current selected values
4. add 2 items (you'll only see 1 item added to values)

> Note; this is unrelated to validation

**Reply by bluebill1049** [reply]
<img width="1191" alt="Screen Shot 2021-11-18 at 10 44 19 pm" src="https://user-images.githubusercontent.com/10513364/142409351-9398b999-1474-4b0a-9f4c-f1fa5da339fa.png">
no, i can't reproduce the issue.

**Reply by ferrybrouwer** [reply]
weird, because i've asked other colleagues to test this as well, they all were able to reproduce this. 
After you select ipad in devtools, you'll need to refresh the browser.

Chrome details:
Version 95.0.4638.69 (Official Build) (x86_64)

![rhf-bug](https://user-images.githubusercontent.com/4455679/142418368-17ba3d3f-ba1a-44a0-ac07-16f5f21e6a35.gif)

**Reply by bluebill1049** [reply]
related to this: https://github.com/react-hook-form/react-hook-form/discussions/7067

**Reply by ferrybrouwer** [reply]
I guess it's related, that's why I've created that issue. Can you reopen the issue?

**Reply by JCalmCrasher** [reply]
Hey @ferrybrouwer, I'm facing this issue and I'm wondering if you were able to fix it, if so how did you fix this?
